File: client/src/app/_models/playerCharacters/playerCharacter.ts
import { Item } from "../item";

export interface PlayerCharacter{
    id: number;
    name: string;
    photoUrl: string;
    level: number;
    currentHitpoints: number;
    maxHitpoints : number;
    gold: number;
    leftHand: Item | undefined;
    rightHand: Item | undefined;
    helmet: Item | undefined;
    body: Item | undefined;
    feet: Item |undefined;
    backPack: (Item | undefined)[] | undefined;
}

File: API/Entities/PlayerCharacter.cs
using System.ComponentModel.DataAnnotations.Schema;
using System.Text.Json.Serialization;

namespace API.Entities;

public class PlayerCharacter
{
    public int Id { get; set; }
    public string Name { get; set; }
    public string PhotoUrl { get; set; }
#nullable enable

    [ForeignKey("Helmet")]
    public int? HelmetItemSaveId { get; set; }
    public ItemSave Helmet { get; set; }

    [ForeignKey("LeftHand")]
    public int? LeftHandItemSaveId { get; set; }
    public ItemSave LeftHand { get; set; }

    [ForeignKey("RightHand")]
    public int? RightHandItemSaveId { get; set; }
    public ItemSave RightHand { get; set; }

    [ForeignKey("Body")]
    public int? BodyItemSaveId { get; set; }
    public ItemSave Body { get; set; }

    [ForeignKey("Feet")]
    public int? FeetItemSaveId { get; set; }
    public ItemSave Feet { get; set; }

    public List<ItemSave> BackPack { get; set; }

    [ForeignKey("AdventureSave")]
    public int? AdventureSaveId { get; set; }
    [JsonIgnore]
    public AdventureSave? AdventureSave { get; set; }
}
